fight/heavy.js

```diff
@@ -0,0 +1,32 @@
+const Soldier = require('./soldier');
+const Unit = require('./unit');
+
+function Heavy() {
+    Unit.apply(this, arguments);
+    this._resist = 20;
+}
+Heavy.prototype = new Soldier();
+
+Heavy.prototype.machineGunAttack = function(unit, noOfShots){
+
+    let bonusAttack, damage = 0, healthBefore, healthAfter;
+
+    for(let index = 0; index < noOfShots; index++) {

```

Comment:
A bit overhead here, dont you think? Maybe there are some ways to not use cycle?

Reply:
for that particular case - yes, we can just multiply attackPower by noOfShots. But in general it depends on how we treat machineGunAttack - as 1 attack or as a number of attacks: attacker could potentially receive XP for each shot and levelup in the middle so remaining shots will have increased damage, same for defender who could levelup and strengthen the defense for example :)

Reply:
Ok, agree, it depends on requirements